keras/optimizers/schedules/learning_rate_schedule.py

```diff
@@ -641,6 +641,12 @@ def __init__(
         self.decay_steps = decay_steps
         self.alpha = alpha
         self.name = name
+        
+        #validating decay_steps argument to raise error if passed
+        #negative values or zero or float values
+        if not isinstance(self.decay_steps,int) or self.decay_steps<=0:
```

Comment:
Note that the decay_steps could be a tensor (from the docstring), so the current check will fail.

Reply:
I think it should be fine to only check the value when it is int, and you can skip the tensor value here.